src/natcap/invest/urban_nature_access.py

```diff
@@ -347,328 +504,647 @@ def execute(args):
         dependent_task_list=[lulc_alignment_task]
     )
 
+    # If we're doing anything with population groups, rasterize the AOIs and
+    # create the proportional population rasters.
+    proportional_population_paths = {}
+    proportional_population_tasks = {}
+    pop_group_proportion_paths = {}
+    pop_group_proportion_tasks = {}
+    if (args['search_radius_mode'] == RADIUS_OPT_POP_GROUP
+            or aggregate_by_pop_groups):
+        aoi_reprojection_task.join()
+        split_population_fields = list(
+            filter(lambda x: re.match(POP_FIELD_REGEX, x),
+                   validation.load_fields_from_vector(
+                       file_registry['reprojected_aois'])))
```

Comment:
It's great for this kind of thing!  Although I'm not sure why I went with `filter` instead of a list comprehension.  Both are pretty nicely functional.